spec/factories/incident.rb

```diff
@@ -9,30 +9,24 @@
       ori           nil
     end
 
-    user { User.find_by_user_id(user_id) || create(:dummy_user) }
+    status { 'in_review' if submit }
 
-    after(:create) do |incident, e|  # e is the FactoryGirl evaluator (has access to transient attributes).
-      incident.screener = create :screener
+    user { User.find_by_user_id(user_id) || create(:dummy_user) }
+    screener
+    general_info { create(:general_info, num_involved_civilians: num_civilians, num_involved_officers: num_officers, ori: (ori || user.ori)) }
 
-      incident.general_info = create(:general_info, ori: e.ori || incident.user.ori) do |g|
-        g.update_attributes(
-          num_involved_civilians: e.num_civilians,
-          num_involved_officers: e.num_officers
-        )
+    # e is the FactoryGirl evaluator (has access to transient attributes).
+    after(:create) do |incident, e|
+      unless e.stop_step == :officers
```

Comment:
Since the order of pages for filling in an incident is Screener -> General Info -> Involved Civilians -> Involved Officers, if the stop step is `:civilians` (that is, the incident stops being filled out at the Involved Civilians page), then *neither* the civilians nor the officers should be created.